### ISSUE TITLE
Mouse can't reach parts of the screen in some LucasArts/Scumm Adventures in windowed mode (mouse = seamless)

### ISSUE THREAD

**Issue by Burrito78** [dosbox-staging/dosbox-staging#1895]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Different version than latest?

DEV

### What Operating System are you using?

Mac OS

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

After starting a new game (e.g. Monkey Island heh) i can't reach the right side of the screen with the cursor (red area).
Instead, the mouse jumps out of the DOSBox window and appears on the host again when reaching the left side of the red box.

Only after moving the cursor to the left edge of the DOSBox window the reachable area gets "reset" somehow to the correct window dimensions and i can reach the full window area with the cursor.

![Bildschirmfoto 2022-09-02 um 14 10 25](https://user-images.githubusercontent.com/61995128/188140725-10448828-02ad-47be-a0dd-b42fb188285a.png)

### Steps to reproduce the behaviour.

Explain how to reproduce
1. Load lates main/dev version
2. Start Monkey Island (for example, no matter which version)
3. Observe how far the mouse cursor goes on the right side of the screen.

### Your configuration

```ini
default
```

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by Burrito78** [comment]
https://anttipeltola.eu/exclude/monkey1windowmetrics1.mp4

**Comment by Grounded0** [comment]
Figured it out myself. Look at that internal cursor position vs where SCUMM places the cursor on starting up the game:

https://anttipeltola.eu/exclude/monkey1windowmetrics2.mp4

**Comment by Burrito78** [comment]
Grounded0 wrote on Discord:

Maybe we can repeat it other SCUMM games

Monkey Island 2: fail
Indiana Jones Fate of the Atlantis: pass
Day of the Tentacle: fail

I think that has something to do with how SCUMM does the initial cursor positioning.

**Comment by Burrito78** [comment]
Also if the window is moved/resized there are suddenly different sized areas, the cursor can‘t reach. Problem is not only on initial start.

**Comment by Grounded0** [comment]
SCUMM games hijack mouse position while Sierra SCI games respect original mouse position:
https://anttipeltola.eu/exclude/monkey1windowmetrics3.mp4

Closing this ticket as wontfix on Saturday 12:00 UTC

**Comment by Burrito78** [comment]
If other DOSBox forks and vanilla show the same behavior and we can't do a workaround easily (i guess we can't) then yes, it's a clearly fine to close this. 

We should add a note to the Wiki about 'Game-Specific Issues' though.

**Comment by Burrito78** [comment]
![Screen Recording 2022-09-03 at 21 12 47](https://user-images.githubusercontent.com/61995128/188288557-f63d3b47-b228-44d6-a2bd-dc5517c62fde.gif)

**Comment by kcgen** [comment]
Had time to look into it.

What I noticed on macOS is that that _relative_ movements of the Lucas Arts cursor are in proportion to SDL's window size versus the host window size.

For example, if the DOS window is half as wide as the host screen, then the Lucas Arts cursor will move half the number of DOSBox pixels versus host pixels.  For @FeralChild64 , this is a bit like using `sensitivity = 50`.

Where as on Linux, it's somewhat the opposite: the proportions are much closer to 1:1, with the Lucas Arts cursor moving slightly more host pixels (a bit like using `sensitivity = 110`).

I should note that on macOS, it reports to SDL that my 4k display is only 1080p, so SDL only believe my ~60% window size is ~1300 pixels, when as it's actually 2600 pixels wide.  Perhaps this reduced ratio is also playing out in the cursor relative distances.

SDL under Linux gets the hosts display resolution exactly as-set.

**Comment by kcgen** [comment]
The game only seems to care about delta (or difference) movements, which means that if your cursor exits and re-enters the window at very different border positions, the game won't "jump" the cursor to the new position on the screen.

For example, if you exit your Lucas Arts cursor from the window's middle-bottom edge (where it will sit motionless), then re-enter your host cursor into the window's middle-right edge, then the Lucas Arts cursor will /not/ jump to the middle-right edge of the window.. instead, it will still be middle-bottom, and actually push over to the left a bit as you re-enter middle-right because the Lucas Arts driver is seing "delta-left" position changes.

This is a telltale sign that seamless won't work, and you should switched to captured.

**Comment by Grounded0** [comment]
Getting seamless to work properly has been a large collective effort and is our showoff feature so i would keep that on. It's also what my emulators that have seamless support default to (SheepShaver etc).

**Comment by Burrito78** [comment]
> The game only seems to care about delta (or difference) movements, which means that if your cursor exits and re-enters the window at very different border positions, the game won't "jump" the cursor to the new position on the screen.
> 
> For example, if you exit your Lucas Arts cursor from the window's middle-bottom edge (where it will sit motionless), then re-enter your host cursor into the window's middle-right edge, then the Lucas Arts cursor will /not/ jump to the middle-right edge of the window.. instead, it will still be middle-bottom, and actually push over to the left a bit as you re-enter middle-right because the Lucas Arts driver is seing "delta-left" position changes.
> 
> This is a telltale sign that seamless won't work, and you should switched to captured.

I tried that now and i have seen something entirely different. If i exit the cursor on the left side of the window and move around the window to the right side and enter from there, the game cursor jumps instantly to the right edge of the screen (green arrow).

<img width="1077" alt="Bildschirmfoto 2022-09-05 um 22 22 29" src="https://user-images.githubusercontent.com/61995128/188510554-87e1a286-b2ee-4d8d-b285-877efc72ed9c.png">

I noticed also that if i enter at different pixel distances from the 0,0-Position of the game window, the offset of the host and game cursor increases linear. It looks as if it is reported wrong to the game. But the question is, why only a handful of LucasArts games react this way?

**Comment by Grounded0** [comment]
SCUMM mouse driver probably. It has been flaky since 1988 and was never fixed.

**Comment by Grounded0** [comment]
Here's a table of SCUMM games you can try to repeat the same errors. Maybe they fixed it at some point:

https://wiki.scummvm.org/index.php/SCUMM/Versions

**Comment by Grounded0** [comment]
Day of the Tentacle seems to track a bit better. You should try The Dig and Full Throttle since they were last SCUMM games for DOS and should have the best mouse driver.

**Comment by Burrito78** [comment]
I tested all LucasArts Adventures and here is the DOSBox Staging seamless mouse compatibility chart:

Maniac Mansion	OK
Zak				OK
Last Crusade		X
LOOM			X
Monkey Island	X
Monkey Island 2	X
Fate of Atlantis	X
DOTT 			X
Sam and Max		X
Full Throttle		OK
The Dig			X

**Comment by kcgen** [comment]
Re-opening to let @FeralChild64 assess it. We know seamless can't work with every game, but maybe there's some valuable information here that he can incorporate. 

For example, on Linux, the exit-enter behavior differs significantly versus what @Burrito78 reports from macOS (it's practically the opposite). Maybe SDL is behaving differently. 


Also, the linearly increasing gap between the host and DOS cursor also differs on Macos versus Linux. 

 So perhaps there's more control or measurements we can gather about the window and the host, and (maybe) get to parity in terms of behaviour across the systems.

**Comment by johnnovak** [comment]
Just to be clear because it's not apparent from the discussion, this is only a problem in windowed mode with seamless mouse pointer enabled? So the games work fine in windowed mode with captured mouse, and in fullscreen, right?

If that's the case, maybe the report should be updated to reflect this (although I realise it's kind of implied because the default config is used).

**Comment by Grounded0** [comment]
Yup that is the case. @Burrito78 can you change ticket title that this only applies to windowed mode.

**Comment by jwrdegoede** [comment]
I just noticed this commit / bug-fix in SDL and I wonder if this is relevant for this issue:

https://github.com/libsdl-org/SDL/commit/cdf312c83d741922ca178b029d11d6176743e3ba

**Comment by kcgen** [comment]
> I just noticed this commit / bug-fix in SDL and I wonder if this is relevant for this issue:
> 
> [libsdl-org/SDL@cdf312c](https://github.com/libsdl-org/SDL/commit/cdf312c83d741922ca178b029d11d6176743e3ba)

Thanks for keeping an eye out in the SDL commits; can you test it? (It would be nice to know that this issue might be solved externally when that version of SDL becomes common in package management systems).

**Comment by Burrito78** [comment]
I will close this issue for now as seamless is no longer our default which makes the issue a rather obscure edge-case, imho.
